### PR TITLE
Migrate README flowchart to Mermaid

### PR DESCRIPTION
**Pull request by stephenswat**
In the spirit of "it's Saturday", I decided to migrate the flowchart of algorithms that we have in our README to Mermaid, which is a graph markup language that recently gained support in Github. It looks pretty and best of all, you can click on links to go places. It's great.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,13 +17,116 @@ Demonstrator tracking chain for accelerators.
...
+The relations between datatypes and algorithms is given in the (approximately
+commutative) diagram shown below. Black lines indicate CPU algorithms, green
+lines indicate CUDA algorithms, and blue lines indicate SYCL algorithms. Solid
+algorithms are ready for use, dashed algorithms are in development or future
+goals. Data types for different heterogeneous platforms are contracted for
+legibility, and identities are hidden.
+
+```mermaid
+flowchart LR
+    subgraph clusterization [<a href='https://github.com/acts-project/traccc/blob/main/core/include/traccc/clusterization/clusterization_algorithm.hpp'>Clusterization</a>]
```
Comment (krasznaa) [inline README.md:29]: Would there be a way of referring to the branch/tag specific version of the files? :thinking:

With GitLab you can use "prefix-less" path names in MarkDown to refer to files/directories in the same branch/tag that you're looking at for the MarkDown file. Is something similar not possible with GitHub/Mermaid?

Of course the idea would be that the documentation in a given tag would stay usable in such a setup, even after things are re-designed in the main branch.

**Review by krasznaa — COMMENTED** [review]
Is it possible to view the rendered image as part of the CI? Unfortunately I didn't look at the actual result yet. :frowning:

File: README.md (hunk 1 above)
Comment (stephenswat) [reply README.md:29]: Sadly that's not possible with Github's current Mermaid integration. :frowning:

**Review by krasznaa — APPROVED** [review]
Let's go with this fancy drawing then. :wink:

**Review by krasznaa — APPROVED** [review]
Re-approving for the bot's benefit...

**Pull request merged**